crates/matrix-sdk/src/event_cache/paginator.rs

```diff
@@ -296,14 +302,17 @@ impl<PR: PaginableRoom> Paginator<PR> {
         let has_prev = response.prev_batch_token.is_some();
         let has_next = response.next_batch_token.is_some();
 
-        *self.prev_batch_token.lock().unwrap() = match response.prev_batch_token {
-            Some(token) => PaginationToken::HasMore(token),
-            None => PaginationToken::HitEnd,
-        };
-        *self.next_batch_token.lock().unwrap() = match response.next_batch_token {
-            Some(token) => PaginationToken::HasMore(token),
-            None => PaginationToken::HitEnd,
-        };
+        {
+            let mut tokens = self.tokens.lock().unwrap();
```

Comment:
I have a deja-vu, didn't I suggest a single lock for this a month or two ago? Anyways, nice that we remove some locks.

Reply:
Yes, that rings a bell, sorry I hadn't listened more :sweat_smile: